sdk/monitor/azquery/custom_client.go

```diff
@@ -27,28 +28,49 @@ type LogsClientOptions struct {
 	azcore.ClientOptions
 }
 
+// LogsClient contains the methods for the LogsClient group.
+// Don't use this type directly, use NewLogsClient() instead.
+type LogsClient struct {
+	host string
+	pl   runtime.Pipeline
+}
+
+// MetricsClient contains the methods for the Metrics group.
+// Don't use this type directly, use NewMetricsClient() instead.
+type MetricsClient struct {
+	host string
+	pl   runtime.Pipeline
+}
+
 // NewLogsClient creates a client that accesses Azure Monitor logs data.
 func NewLogsClient(credential azcore.TokenCredential, options *LogsClientOptions) *LogsClient {
 	if options == nil {
 		options = &LogsClientOptions{}
 	}
-	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{"https://api.loganalytics.io/.default"}, nil)
+
+	host := cloud.AzurePublic.Services[monitorQueryLogs].Endpoint
+	if c, ok := options.Cloud.Services[monitorQueryLogs]; ok {
+		host = c.Endpoint
```

Comment:
This elides an error condition: a user may set `options.Cloud.Services[monitorQueryLogs]` but not that value's `Endpoint` field. In that case, it appears the user intends, but has failed, to provide private cloud configuration; what should the constructor do? This is why we return an error from ARM client constructors: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/runtime/pipeline.go#L57

@jhendrixMSFT another instance where we'd be in a jam if the module were GA already

Reply:
I was wondering about this but didn't look deep enough.  Makes me think that we should update the guidelines to have ctors return errors by default.

Reply:
Yeah. Some, but not all, constructors need to return errors, so we have to choose consistency or brevity. I keep forgetting whether Key Vault constructors return errors, so I think we should favor consistency. When last we discussed this, I believe @JeffreyRichter was open to returning errors across the SDK?

Reply:
Yes, for client ctors we can always return an error.

Reply:
Updated constructors to return an error in commit [3772df1](https://github.com/Azure/azure-sdk-for-go/pull/19462/commits/3772df1fc545f3ba93ce4eb8c7ba45c582aff39d)